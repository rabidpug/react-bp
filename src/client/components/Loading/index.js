// @flow

import React from 'react';

const Loading: Function = () => ( <div>
  {'Loading...'}
</div> );

export default Loading;
