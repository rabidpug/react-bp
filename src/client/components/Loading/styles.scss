.loading-content {
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

.spinner {
  max-height: inherit !important;
  min-height: 250px;
}
