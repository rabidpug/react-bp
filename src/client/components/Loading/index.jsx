// @flow

import { Icon, Spin, } from 'antd';

import React from 'react';
import styles from 'Styles/Loading';

const antIcon = ( <Icon
  spin
  style={ { fontSize: 24, } }
  type='loading' /> );
const Loading: Function = () => (
  <Spin
    className={ styles.spinner }
    delay={ 500 }
    indicator={ antIcon }
    size='large'
    tip='Loading...'
    wrapperClassName={ styles.centrePosition }>
    <div />
  </Spin>
);

export default Loading;
