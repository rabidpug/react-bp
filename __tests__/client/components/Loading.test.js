import Loading from 'Components/Loading';
import React from 'react';
import { shallow, } from 'enzyme';

describe(
  'Loading', () => {
    it(
      'Should render the Loading', () => {
        const wrapper = shallow( <Loading /> );

        expect( wrapper ).toMatchSnapshot();
      }
    );
  }
);
