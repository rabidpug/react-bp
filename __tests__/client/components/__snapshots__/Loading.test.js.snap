// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Should render the Loading 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div />,
      "className": "spinner",
      "delay": 500,
      "indicator": <Icon
        spin={true}
        style={
          Object {
            "fontSize": 24,
          }
        }
        type="loading"
      />,
      "prefixCls": "ant-spin",
      "size": "large",
      "spinning": true,
      "tip": "Loading...",
      "wrapperClassName": "loadingContent",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div />,
        "className": "spinner",
        "delay": 500,
        "indicator": <Icon
          spin={true}
          style={
            Object {
              "fontSize": 24,
            }
          }
          type="loading"
        />,
        "prefixCls": "ant-spin",
        "size": "large",
        "spinning": true,
        "tip": "Loading...",
        "wrapperClassName": "loadingContent",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
